### DISCUSSION TITLE
[mpvnet] 播放时自动全屏、关联图片和自动加载

### DISCUSSION THREAD

**Discussion by xiaozixun** [hooke007/mpv_PlayKit#43] [libmpv]
我想让它在不播放文件的时候时默认窗口，播放文件时就自动全屏。可我找了找似乎没发现这个设置，而原版文档我用机翻大致看了下也没有找到。
请问这个该如何设置？

关联文件时视频和音乐没有问题，但图片却没有反应，进入我电脑系统的默认应用也没有mpv这个选项。
![批注 2021-11-12 194836](https://user-images.githubusercontent.com/22409452/141462547-7273585a-596d-4d41-b9ae-8954e5408e6f.png)

自动加载是启用了auto-load-folder后能否让视频单个加载播放，而音乐和图片则是列表播放？也就是全部（列表）重复和单个重复的设置。

**Comment by hooke007** [comment]
>不播放文件的时候时默认窗口，播放文件时就自动全屏

用profile能实现后半部分
```
[fs-playing]
 profile-cond    = not pause
 profile-restore = copy
 fs
```

**Reply by xiaozixun** [reply]
> 
> 
> > 不播放文件的时候时默认窗口，播放文件时就自动全屏
> 
> 用profile能实现后半部分
> 
> ```
> [fs-playing]
>  profile-cond    = not pause
>  profile-restore = copy
>  fs
> ```

profiles.conf吗？我写入后还是和之前一样。

**Reply by xiaozixun** [reply]
是我错设置了什么吗？麻烦看下：
mpv.conf：
`border = no
loop-playlist = inf
osd-on-seek = msg-bar
volume = 100
audio-file-auto = fuzzy
sub-auto = fuzzy
screenshot-jpeg-quality = 100
screenshot-webp-quality = 100
hwdec = auto
autofit = 100
osd-fractions = yes
save-position-on-quit = yes
osc = no`

mpvnet.conf：
`# Editor
auto-play = yes
process-instance = queue
auto-load-folder = no
start-size = video
`

profiles.conf：
`[extension.vpy]
 demuxer-lavf-format = vapoursynth
[fs-playing]
 profile-cond = not pause
 profile-restore = copy
 fs`

**Reply by hooke007** [reply]
。。。我好像当前版本漏掉了一个默认参数，现在先写 mpv.conf 里的末尾

**Comment by hooke007** [comment]
auto-load-folder 是把当前目录文件全部装进播放列表，后半句话没懂。循环播放loop？

**Reply by xiaozixun** [reply]
> 
> 
> auto-load-folder 是把当前目录文件全部装进播放列表，后半句话没懂。循环播放loop？

就是其它播放器上的单文件循环播放和播放列表内全部文件播放，或者说就是“单曲循环”和“列表循环”（这样不知道解释清楚了没有）

**Reply by hooke007** [reply]
https://github.com/hooke007/MPV_lazy/discussions/42

参考这个针对视频类文件单独设定 profile

核心参数是 `--loop-file=inf`

**Reply by hooke007** [reply]
https://github.com/mpv-player/mpv/wiki/FAQ#How_do_I_create_a_Conditional_auto_profile_that_applies_to_all_videos_or_all_audio

mpv官方给的针对视频类型的判定条件示例

**Reply by xiaozixun** [reply]
> 
> 
> #42
> 
> 参考这个针对视频类文件单独设定 profile
> 
> 核心参数是 `--loop-file=inf`

所以写法就是`[video]
profile-cond=get("loop-file") == inf`吗？
还是
`[extension.mkv,.mp4,.wmv] 
  --loop-file=inf`

**Reply by hooke007** [reply]
。。。。我给的示例写起来都比较麻烦（一个要堆条件，一个要写一堆profile），你就按照官方的条件写

```
[video]
 profile-cond    = get("current-tracks/video/albumart") == false
 profile-restore = copy
 loop-file       = inf
```

**Reply by hooke007** [reply]
用我的土条件也不是不行
```
[extension.mkv]
 loop-file = inf

[extension.mp4]
 loop-file = inf

[extension.wmv]
 loop-file = inf

[extension.flv]
 loop-file = inf

...
```
或者
```
[video]
 profile-cond    = filename:match("%.mkv$")~=nil or filename:match("%.mp4$")~=nil filename:match("%.wmv$")~=nil or filename:match("%.flv$")~=nil
 profile-restore = copy
 loop-file       = inf
```

**Reply by xiaozixun** [reply]
> 
> 
> 。。。。我给的示例写起来都比较麻烦（一个要堆条件，一个要写一堆profile），你就按照官方的条件写
> 
> ```
> [video]
>  profile-cond    = get("current-tracks/video/albumart") == false
>  profile-restore = copy
>  loop-file       = inf
> ```

谢谢，成功了。不过又产生了个新问题：浏览图片时跳转下一张图太快了，怎么控制下？还有就是我用了`Thumbnailer_Worker.lua``Thumbnailer.lua`和`osc_lazy.lua`（其它的还有`playlistmanager.lua`和`cycle_adevice.lua`）也用了osc = no了，却不显示略缩图，不知道是怎么回事。

**Reply by hooke007** [reply]
> 浏览图片时跳转下一张图太快

https://mpv.io/manual/master/#options-image-display-duration

> 不显示略缩图

没有找到ffmpeg

**Reply by xiaozixun** [reply]
> 
> 没有找到ffmpeg

我在[https://www.gyan.dev/ffmpeg/builds/](url)下了ffmpeg-git-essentials.7z，然后解压到源目录里。
请问是哪步出了问题？

**Reply by hooke007** [reply]
解压到哪里都行，问题是系统不知道，mpv也不知道

要么让系统知道 添加到环境变量；

要么让mpv知道
https://github.com/hooke007/MPV_lazy/blob/ada5ba31763a53123473465d29fedfd9dd9c4268/portable_config/script-opts/thumbnailer.conf#L17-L18

**Reply by xiaozixun** [reply]
> 要么让系统知道 添加到环境变量；

![批注 2021-11-12 230659](https://user-images.githubusercontent.com/22409452/141488495-e8b0b523-0dba-4a6c-9c14-2f86a5b12072.png)


> 要么让mpv知道

![批注 2021-11-12 23019](https://user-images.githubusercontent.com/22409452/141488732-60e106f9-0a2d-4d53-bda8-3a7bcd983e6b.png)


都失败了，是我哪里没写对吗

**Reply by hooke007** [reply]
添加到变量之后看看终端输入 `ffmpeg` 会输出什么

**Reply by hooke007** [reply]
我看你的第二幅图，，，你的脚本设置文件总不是照抄没改吧。。。懒人包里默认是关闭生成和显示的
https://github.com/hooke007/MPV_lazy/blob/ada5ba31763a53123473465d29fedfd9dd9c4268/portable_config/script-opts/thumbnailer.conf#L5-L7

**Reply by xiaozixun** [reply]
> 我看你的第二幅图，，，你的脚本设置文件总不是照抄没改吧。。。懒人包里默认是关闭生成和显示的
> 
> https://github.com/hooke007/MPV_lazy/blob/ada5ba31763a53123473465d29fedfd9dd9c4268/portable_config/script-opts/thumbnailer.conf#L5-L7

这边改成过yes但无效，我又改回来了。
我后天有空再实验吧（明后两天有事，可能顾不上），电脑已经关了，现在是手机回。
很感谢您的帮助。

**Reply by hooke007** [reply]
必须两个yes ＋ ffmpeg存在path或选项，有空回即可。

**Reply by xiaozixun** [reply]
> 
> 
> 必须两个yes ＋ ffmpeg存在path或选项，有空回即可。

原来是16行也得写入才行……

我想问下发行说明里的“不要使用覆盖旧版本文件的形式进行更新”是执行一次注册表信息清理与重置就行吗？

此外我想问下osc_lazy开启后怎么让播放列表不会在切换文件时一同显示？与用mpv播放流媒体除了[https://github.com/stax76/mpv.net/issues/119](url)和youtube-dl外还有没有其它办法？

最后就是官方文档有中文版的吗，英文查找太痛苦。

**Reply by hooke007** [reply]
> 我想问下发行说明里的“不要使用覆盖旧版本文件的形式进行更新”是执行一次注册表信息清理与重置就行吗？

这是两回事。
不要使用覆盖旧版本文件的形式进行更新：删掉所有旧文件
注册表信息清理与重置：这是自带的一个bat，重置设置和清理缓存用的

> 此外我想问下osc_lazy开启后怎么让播放列表不会在切换文件时一同显示？

不懂，附图说明

> 与用mpv播放流媒体除了[https://github.com/stax76/mpv.net/issues/119](url)和youtube-dl外还有没有其它办法？

说法不成立。。。mpv使用ytdl解析http地址，你想表达的是不是如何用mpvnet加载网址。右键菜单就有导入网址

> 最后就是官方文档有中文版的吗，英文查找太痛苦。

无

**Reply by xiaozixun** [reply]
> 
> 
> 这是两回事。 不要使用覆盖旧版本文件的形式进行更新：删掉所有旧文件 注册表信息清理与重置：这是自带的一个bat，重置设置和清理缓存用的
好的，谢谢。
> 不懂，附图说明
![批注 2021-11-14 131312](https://user-images.githubusercontent.com/22409452/141668560-fbb450cd-1a82-41e8-b43e-adcbf33c83d3.png)
就这个，虽然我在不知道的情况下改让它不会再切换下一个文件播放时显示了
> 说法不成立。。。mpv使用ytdl解析http地址，你想表达的是不是如何用mpvnet加载网址。右键菜单就有导入网址
我的目的是让浏览器使用mpvnet来播放网页上的视频。可右键菜单导入的话，mpvnet并不会识别网页链接（如https://www.bilibili.com/video/BV1K34y1Z75t）；如果在浏览器的控制台里找也十分麻烦而且有些网站只会给个blob地址；而ytdl本质上是个下载工具（我看的机翻，应该没理解错吧），我不太需要它。
> 无
好吧

**Reply by zhongfly** [reply]
……你没理解，如果是用ytdl，你输入B站链接，它就能分析得到实际的视频文件链接用于播放
另外你也需要理解，在线视频播放的本质也是下载视频片段进行播放。

**Reply by xiaozixun** [reply]
> 
> 
> ……你没理解，如果是用ytdl，你输入B站链接，它就能分析得到实际的视频文件链接用于播放 另外你也需要理解，在线视频播放的本质也是下载视频片段进行播放。

不是的，我的最终目的其实是用来测试浏览器的。本身我只需要观看本地媒体而并没有观看流媒体的需求，只不过是我使用的一个浏览器最近有些不大对，所以才想着能不能拿本地播放器来测试的。
故此基于以上的原因，我并不想用ytdl。

**Reply by hooke007** [reply]
你还是没理解。。。ytdl被广泛的第三方播放器作为网址解析工具，比如mpc/pot，他们解析网址播放也是依赖的ytdl

**Reply by hooke007** [reply]
<s>测完了。这不是脚本的问题，是上游的bug，原始osc就有这个bug</s> 这是自带的功能不是bug

**Reply by zhongfly** [reply]
> 不是的，我的最终目的其实是用来测试浏览器的。本身我只需要观看本地媒体而并没有观看流媒体的需求，只不过是我使用的一个浏览器最近有些不大对，所以才想着能不能拿本地播放器来测试的。 故此基于以上的原因，我并不想用ytdl。

"我的目的是让浏览器使用mpvnet来播放网页上的视频。可右键菜单导入的话，mpvnet并不会识别网页链接（如https://www.bilibili.com/video/BV1K34y1Z75t）" 简而言之，用ytdl才能达成你的目的，不用就不能

**Reply by xiaozixun** [reply]
> 
> 
> 你还是没理解。。。ytdl被广泛的第三方播放器作为网址解析工具，比如mpc/pot，他们解析网址播放也是依赖的ytdl

所以说我想要测试的话就必须使用ytdl？好吧，我回去看看的。

**Reply by xiaozixun** [reply]
> 
> 
> osc_lazy应该没有按键会关联这个播放列表脚本，这个不清楚什么情况。（我可以复现这个问题
> 
> 测完了。这不是脚本的问题，是上游的bug，原始osc就有这个bug

哦，我倒是在osc_lazy和osc的配置里胡乱设置，莫名其妙地把它关了……

**Reply by xiaozixun** [reply]
> 
> 
> > 不是的，我的最终目的其实是用来测试浏览器的。本身我只需要观看本地媒体而并没有观看流媒体的需求，只不过是我使用的一个浏览器最近有些不大对，所以才想着能不能拿本地播放器来测试的。 故此基于以上的原因，我并不想用ytdl。
> 
> "我的目的是让浏览器使用mpvnet来播放网页上的视频。可右键菜单导入的话，mpvnet并不会识别网页链接（如[https://www.bilibili.com/video/BV1K34y1Z75t）](https://www.bilibili.com/video/BV1K34y1Z75t%EF%BC%89)" 简而言之，用ytdl才能达成你的目的，不用就不能

嗯，我去看看。
同样也很感谢你的帮助。

**Reply by hooke007** [reply]
> 哦，我倒是在osc_lazy和osc的配置里胡乱设置，莫名其妙地把它关了……

传一下你的相关设置我看看

**Reply by xiaozixun** [reply]
> 
> 
> > 哦，我倒是在osc_lazy和osc的配置里胡乱设置，莫名其妙地把它关了……
> 
> 传一下你的相关设置我看看

`https://cowtransfer.com/s/3cadccf5d1cc44`
找了个文件分享网站，playlistmanager.conf我并没有修改过，所以没有上传

**Reply by hooke007** [reply]
。。。这里底下就可以直接传文件不用这么复杂
![Snipaste_2021-11-14_20-01-00](https://user-images.githubusercontent.com/41094733/141680016-45e6842d-393c-4b28-84c7-5de07cd3ec8e.jpg)

**Reply by xiaozixun** [reply]
> 
> 
> 。。。这里底下就可以直接传文件不用这么复杂 ![Snipaste_2021-11-14_20-01-00](https://user-images.githubusercontent.com/41094733/141680016-45e6842d-393c-4b28-84c7-5de07cd3ec8e.jpg)

我还以为只能传图片……
不过那个网站能下载吗？需不需要我再传一次？

**Reply by hooke007** [reply]
不需要。。。这个问题挺诡异，无法复现你的描述 `莫名其妙地把它关了`

**Reply by xiaozixun** [reply]
> 
> 
> 不需要。。。这个问题挺诡异，无法复现你的描述

https://cowtransfer.com/s/a69e91a41b7844

截了张gif，结果太大传不上来，只能用之前的网站。

不过奇怪的是我重新试了试发现快捷键切换已经不会出现了，只有按键才会。

**Reply by hooke007** [reply]
快捷键切换本来就不会出现，osc按钮切换会出现。

**Reply by xiaozixun** [reply]
> 
> 
> 快捷键切换本来就不会出现，osc按钮切换会出现。

原来是这样，谢谢

**Comment by hooke007** [comment]
关联图片我 win10 和11都测试正常，无法复现，实在不行就手动指定格式

**Reply by xiaozixun** [reply]
非常感谢您对我的细致帮助与耐心回复，辛苦了。
只不过我明后天还有事，现在无法再继续了。

**Comment by zhongfly** [comment]
![image](https://user-images.githubusercontent.com/11155705/141671958-5a584c9e-6a1d-4f63-a532-906409491843.png)
mpv使用yt-dlp看B站视频，就是直接右键菜单导入的，你只需要在mpvnet文件夹内放好yt-dlp.exe，就可以识别像B站这样的网页链接

真的建议你试过之后再问，而不是武断的认为“ytdl本质上是个下载工具（我看的机翻，应该没理解错吧），我不太需要它。”

**Reply by zhongfly** [reply]
不过我觉得体验并不好，还是在网页上看更好

**Comment by zhongfly** [comment]
切换文件时会显示播放列表的问题，我一直以为是feature（“快捷键切换本来就不会出现，按键切换会出现。”那看来的确是
一直存在，最简配置下依旧复现，只修改mpv.conf里的osc=no，仅使用osc_lazy

点击osc上的下一个文件后，切换文件时必定出现播放列表
（基于全新下载的mpv.net 5.5.0.4便携版）
[portable_config.zip](https://github.com/hooke007/MPV_lazy/files/7533835/portable_config.zip)

**Reply by hooke007** [reply]
是feature啊，原版就是这样

**Reply by xiaozixun** [reply]
> 
> 
> 切换文件时会显示播放列表的问题，我一直以为是feature（“快捷键切换本来就不会出现，按键切换会出现。”那看来的确是 一直存在，最简配置下依旧复现，只修改mpv.conf里的osc=no，仅使用osc_lazy
> 
> 点击osc上的下一个文件后，切换文件时必定出现播放列表 （基于全新下载的mpv.net 5.5.0.4便携版） [portable_config.zip](https://github.com/hooke007/MPV_lazy/files/7533835/portable_config.zip)

原来是这样，谢谢

**Reply by hooke007** [reply]
这个功能得到了改进
https://github.com/mpv-player/mpv/commit/b8926dd4840d3612065c75d51a2f6b0fd96936fa
